cmd/otel-allocator/watcher/file.go

```diff
@@ -43,19 +43,20 @@ func NewFileWatcher(logger logr.Logger, config config.CLIConfig) (*FileWatcher,
 
 	return &FileWatcher{
 		logger:         logger,
-		configFilePath: *config.ConfigFilePath,
+		configFilePath: configFilePath,
 		watcher:        fileWatcher,
 		closer:         make(chan bool),
 	}, nil
 }
 
 func (f *FileWatcher) LoadConfig(_ context.Context) (*promconfig.Config, error) {
-	cfg, err := config.Load(f.configFilePath)
+	cfg := config.CreateDefaultConfig()
+	err := config.LoadFromFile(f.configFilePath, &cfg)
```

Comment:
Would this result in us overriding the flags someone sets? Maybe no because I don't think they can set anything for the promConfig?

Reply:
I'd like to deprecate as many flags as I can in a later step because of ambiguities like this. Maybe even get rid of this hot reload functionality, as it's a bit confusing in the context of K8s. It's an antipattern to let changes to a ConfigMap affect a Pod without restarting it - prometheus-operator mostly does it because restarting Prometheus is somewhat expensive, but that's not really a problem for TA.